tsdb/head_read.go

```diff
@@ -325,11 +325,25 @@ func (s *memSeries) chunk(id chunks.HeadChunkID, chunkDiskMapper *chunks.ChunkDi
 	if ix < 0 || ix > len(s.mmappedChunks) {
 		return nil, false, storage.ErrNotFound
 	}
+	mc := memChunkPool.Get().(*memChunk)
```

Comment:
Now we get it from the pool even if `cPool == nil`. I would move this back to the previous place and call it also from the following `if` branch, so that we call it only when strictly required.